_format/1.1/index.md

```diff
@@ -1781,6 +1807,122 @@ parameter from this specification, it **MUST** return `400 Bad Request`.
+
+A profile **MAY** assign meaning to particular elements of the document
+structure, such as resource attributes or members of meta objects, and even
+reserved query parameters, such as `filter`. The scope of a profile **MUST** be
+clearly delineated. The elements reserved by a profile, and the meaning assigned
+to those elements, **MUST NOT** change over time or else the profile **MUST** be
+considered a new profile with a new URI. However, a profile **MAY** evolve
+additively within the scope originally claimed by the profile.
+
+For example, let's say that a profile reserves a `timestamps` member in the
+`meta` object of every resource. Originally, this profile defines the value
+of `timestamps` as an object that must contain one member: `created`. The
+profile could evolve to allow an optional member, `updated`, in the `timestamps`
+object. But it could not make that member required, nor could it introduce
+a new sibling to `timestamps`.
+
+### <a href="#profile-keywords" id="profile-keywords" class="headerlink"></a> Profile Keywords
+
+A profile **SHOULD** explicitly declare "keywords" for any elements that it
+introduces to the document structure. If a profile does not explicitly declare a
+keyword for an element, then the name of the element itself can be considered a
+keyword. Keywords **MAY** be aliased in any representation through the use of
+[profile descriptors](profile-descriptors), as described below.
+
+### <a href="#profile-media-type-parameter" id="profile-media-type-parameter" class="headerlink"></a> `profile` Media Type Parameter
+
+The `profile` media type parameter is used to describe the application of
+one or more profiles to the JSON API media type. The value of the `profile`
+parameter **MUST** equal a whitespace-separated list of profile URIs.
+
+Clients and servers **SHOULD** use the `profile` media type parameter in
+conjunction with the JSON API media type in a `Content-Type` header to specify
+the application of one or more profiles to a JSON API document.
+
+A client **MAY** use the `profile` media type parameter in conjunction with the
+JSON API media type in an `Accept` header to _request_, but not _require_, that
+the server apply one or more profiles to the response. When such a request is
+received, a server **SHOULD** attempt to apply the requested profiles in
+processing a response.
+
+### <a href="#profile-query-parameter" id="profile-query-parameter" class="headerlink"></a> `profile` Query Parameter
+
+A client **MAY** use the `profile` query parameter to _require_ the application
+of one or more profiles to the JSON API media type in processing a request. The
+value of the `profile` query parameter **MUST** equal a URI-encoded
+whitespace-separated list of profile URIs.
+
+If a server receives a request requiring the application of a profile or
+combination of profiles that it can not apply, it **MUST** respond with a `400
+Bad Request` status code.
```

Comment:
The fact that the `profile` query param has different semantics than the media type parameter seems a bit confusing to me. Are there any other reasons for this besides the need of having ways for expressing both requesting and requiring a profile? It might be an option to somehow encode these semantics (request vs. require) in the query/media type parameter that's passed vs. where it's passed (header vs. query param)?

Reply:
@marcoow We had quite the discussion in #1207 that largely centered upon this matter, and it took me a while to accept this approach, so I certainly understand your perspective. Please see the [note](https://github.com/json-api/json-api/pull/1268/commits/510e783b54917ed15b13931dd07dff0e1d552f27#diff-571f6cee9a492e0e177801c21a4b2b2aR1896) that @ethanresnick just added to expand upon the difference and let us know if this clarifies the matter for you.